README.md

```diff
@@ -354,6 +354,51 @@ pip install jina[hub] && jina hub new --type app
 
 This creates a Python entrypoint, YAML configs and a Dockerfile. You can start from there.
 
+## Multi & Cross Modality
+
+Jina is a data type agnostic framework. It basically can work with any data type. 
+One just needs to find a way to project the documents of this type into an embedding space.
+
+But Jina does not only offer this but also allows you to run cross and multi-modal search flows. 
+To better understand what this implies we first need to understand the concept of modality. 
+
+One may think that different modalities correspond to different kinds of data (images and text in this case).
+However, this is not accurate. For example, one can do cross-modal search by searching images from different points of view, 
+or searching for matching titles for given paragraph text.
+Therefore, one can consider that a modality is related to a given data distribution from which input may come. 
+
+For this reason, and to have first-class support for cross and multi-modal search, Jina offers modality as an attribute in its Document primitive type.
+Now that we are agreed on the concept of modality, we can describe cross-modal and multi-modal search.
+ 
+ - Cross-modal search can be defined as a set of retrieval applications that try to effectively find relevant documents of modality A by querying with documents from modality B.
+ - Multi-modal search can be defined as a set of retrieval applications that try to effectively project documents of different modalities into a
+ common embedding space. 
```

Comment:
into a common embedding space and find relevant document with respect to the fusion of multiple modalities